cli/context/store/store.go

```diff
@@ -259,12 +271,111 @@ func Export(name string, s Reader) io.ReadCloser {
 	return reader
 }
 
+type fileData interface {
+	get() ([]byte, error)
+}
+
+type zipFileData struct {
+	fileData
+	zf *zip.File
+}
+
+func (zfd *zipFileData) get() ([]byte, error) {
+	src, err := zfd.zf.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer src.Close()
+	data, err := ioutil.ReadAll(src)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+type tarFileData struct {
+	fileData
+	tr io.Reader
+}
+
+func (tfd *tarFileData) get() ([]byte, error) {
+	data, err := ioutil.ReadAll(tfd.tr)
```

Comment:
We might want to protect this against huge files being passed and being read into memory (also see https://github.com/docker/cli/pull/1867/files#r285028468)

Reply:
I've added `checkFileSize` for individual files both for `.tar` & `.zip`. Let me know if it seems satisfactory for you, or if you were envisioning a different implementation.

Reply:
Could use `io.LimitReader`; https://golang.org/pkg/io/#LimitReader

Reply:
If we want it to produce an error (instead of silently stopping at the limit), we could use this approach: https://github.com/docker/distribution/pull/2340 (but that may be a bit overcomplicating things)

Reply:
Well if we in fact go with the `io.LimitReader` route, I'd have to implement the error handling as well because since reader will fail silently we might get back errors such as `invalid json object` of some sorts that will just result in red herrings.

Reply:
I've added `LimtiedReadAll` as a part of `io_utils`, check it out, I think it covers what we are trying to achieve here pretty well. I am going to resolve this as it's outdated, please feel free to add additional comments on the new set of functionality.